homeassistant/components/twinkly/light.py

```diff
@@ -0,0 +1,223 @@
+"""The Twinkly light component."""
+
+import logging
+from typing import Any, Dict, Optional
+
+from aiohttp import ClientError
+from twinkly_client import TwinklyClient
+import voluptuous as vol
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    PLATFORM_SCHEMA,
+    SUPPORT_BRIGHTNESS,
+    LightEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_NAME
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.config_validation import string
+from homeassistant.helpers.typing import HomeAssistantType
+
+from .const import (
+    ATTR_HOST,
+    CONF_ENTRY_HOST,
+    CONF_ENTRY_ID,
+    CONF_ENTRY_MODEL,
+    CONF_ENTRY_NAME,
+    DEV_MODEL,
+    DEV_NAME,
+    DOMAIN,
+    HIDDEN_DEV_VALUES,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+# Schema for configuration.yaml
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_HOST): string,
+        vol.Optional(CONF_NAME): string,
+    }
+)
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
```

Comment:
Platform setup is deprecated. You also don't need it as you have a config flow.

Reply:
I've kept it only for backward compatibility with the custom_component. I'll remove it.